processing/src/main/java/org/apache/druid/query/aggregation/any/FloatAnyVectorAggregator.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.query.aggregation.any;
+
+import org.apache.druid.segment.vector.VectorValueSelector;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Vectorized implementation of the {@link FloatAnyBufferAggregator}
+ */
+public class FloatAnyVectorAggregator extends NumericAnyVectorAggregator
+{
+  public FloatAnyVectorAggregator(VectorValueSelector vectorValueSelector)
+  {
+    super(vectorValueSelector);
+  }
+
+  @Override
+  void initValue(ByteBuffer buf, int position)
+  {
+    buf.putFloat(position, 0F);
+  }
+
+  @Override
+  boolean putAnyValueFromRow(ByteBuffer buf, int position, int startRow, int endRow)
+  {
+    float[] values = vectorValueSelector.getFloatVector();
+    boolean isRowsWithinIndex = startRow < endRow && startRow < values.length;
```

Comment:
by the way, are these checks necessary? The framework calling aggregators should be taking care of passing valid ranges. If not, its a bug and exception will be raised. I didn't notice these checks in other vector aggregation implementations.

Reply:
I wasn't sure if it's needed or not. The other aggregators have these checks implicitly within the for loops I think